src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Copies the emails of all persons in the address book to the clipboard.
+ */
+public class CopyCommand extends Command {
+
+    public static final String COMMAND_WORD = "copy";
+    public static final String MESSAGE_SUCCESS = "Copied emails of all listed persons to clipboard";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
```

Comment:
I see that you are copying names that are already shown in the GUI

Reply:
Thanks! I've tried keeping to the names in the GUI to keep things simple.